src/main/java/com/artipie/api/artifactory/FromRqLine.java

```diff
@@ -58,14 +90,53 @@
      */
     Optional<String> get() {
         final Optional<String> username;
-        final Matcher matcher = UserFromRqLine.PTRN.matcher(
+        final Matcher matcher = this.ptrn.matcher(
             new RequestLineFrom(this.rqline).uri().toString()
         );
         if (matcher.matches()) {
-            username = Optional.of(matcher.group("username"));
+            username = Optional.of(matcher.group(this.mtchgroup.group()));
         } else {
             username = Optional.empty();
         }
         return username;
     }
+
+    /**
+     * Group name for part of the request line.
+     */
+    enum Group {
```

Comment:
@genryxy as you created enum, let's add patterns to this enum, create getter for the patterns and rename it to `Pattern` or `RqPattern`:
```
USER(Pattern.compile("/api/security/users/(?<username>[^/.]+)"))
...
public Pattern pattern() {
   return this.pattern;
}
```
Thus we will get rid of constants and will use enum items in `ArtipieApi`. 

From `FromRqLine.get()` method can address matcher group by index instead of group name, group names will not be neede.